frontend/src/metabase/query_builder/components/notebook/steps/JoinStep.jsx

```diff
@@ -537,6 +545,16 @@ function getDimensionSourceName(dimension) {
     .displayName();
 }
 
+function getDimensionDisplayName(dimension) {
+  if (!dimension) {
+    return t`Pick a column...`;
+  }
+  if (dimension.temporalUnit()) {
+    return `${dimension.displayName()}: ${dimension.subDisplayName()}`;
+  }
+  return dimension.displayName();
```

Comment:
would it be awful to make this something like...
```
return [
  dimension.displayName(),
  dimension.subDisplayName()
].filter(str => str !== "").join(': ');
```
so that we can avoid have dimension specific logic here?

Reply:
You're right it'd be better to hide this logic somewhere, the problem is `subDisplayName` returns `"Default"` for most of the dimensions. And for numeric ones, it would return the binning strategy. So a join sections could look like the following:

<img width="837" alt="CleanShot 2021-09-16 at 13 09 59@2x" src="https://user-images.githubusercontent.com/17258145/133594146-1f996487-fbc6-445d-9194-309e90ebcae8.png">